src/test/java/com/google/api/codegen/testdata/php_test_library.baseline

```diff
@@ -0,0 +1,823 @@
+
+        $this->assertEquals($formattedName, $actualRequestObject->getName());
+
+        $this->assertEquals($expectedResponse, $response);
+
+        $this->assertTrue($grpcStub->isExhausted());
+    }
+
+    /**
+     * @test
+     */
+    public function getBookFromAnywhereTest()
+    {
+        $grpcCredentialsHelper = new GrpcCredentialsHelper([]);
+        $createGrpcStub = function ($hostname, $opts) {
+            return new MockLibraryServiceImpl($hostname, $opts);
+        };
+        $grpcStub = $grpcCredentialsHelper->createStub(
+            $createGrpcStub,
+            LibraryServiceClient::SERVICE_ADDRESS,
+            LibraryServiceClient::DEFAULT_SERVICE_PORT
+        );
+        $client = new LibraryServiceClient(['createLibraryServiceStubFunction' =>
+            function ($hostname, $opts) use ($grpcStub) {
+                return $grpcStub;
+            },
+        ]);
+
+        $this->assertTrue($grpcStub->isExhausted());
+
+        // Mock response
+        $name2 = "name2-1052831874";
+        $author = "author-1406328437";
+        $title = "title110371416";
+        $read = true;
+        $expectedResponse = new BookFromAnywhere();
+        $expectedResponse->setName($name2);
+        $expectedResponse->setAuthor($author);
+        $expectedResponse->setTitle($title);
+        $expectedResponse->setRead($read);
+        $grpcStub->addResponse($expectedResponse);
+
+        // Mock request
+        $formattedName = LibraryServiceClient::formatBookName("[SHELF_ID]", "[BOOK_ID]");
+        $formattedAltBookName = LibraryServiceClient::formatBookName("[SHELF_ID]", "[BOOK_ID]");
+
+        $response = $client->getBookFromAnywhere($formattedName, $formattedAltBookName);
+        $actualRequests = $grpcStub->getReceivedCalls();
+        $this->assertSame(1, count($actualRequests));
+        list($actualFuncCall, $actualRequestObject) = $actualRequests[0];
+        $this->assertSame('GetBookFromAnywhere', explode('/', $actualFuncCall)[2]);
+
+        $this->assertEquals($formattedName, $actualRequestObject->getName());
+        $this->assertEquals($formattedAltBookName, $actualRequestObject->getAltBookName());
+
+        $this->assertEquals($expectedResponse, $response);
+
+        $this->assertTrue($grpcStub->isExhausted());
+    }
+
+    /**
+     * @test
+     */
+    public function updateBookIndexTest()
+    {
+        $grpcCredentialsHelper = new GrpcCredentialsHelper([]);
+        $createGrpcStub = function ($hostname, $opts) {
+            return new MockLibraryServiceImpl($hostname, $opts);
+        };
+        $grpcStub = $grpcCredentialsHelper->createStub(
+            $createGrpcStub,
+            LibraryServiceClient::SERVICE_ADDRESS,
+            LibraryServiceClient::DEFAULT_SERVICE_PORT
+        );
+        $client = new LibraryServiceClient(['createLibraryServiceStubFunction' =>
+            function ($hostname, $opts) use ($grpcStub) {
+                return $grpcStub;
+            },
+        ]);
```

Comment:
This section that creates the stub & client is repeated in every test - maybe make a helper function or setup function that can create the pair of them?